src/main/java/com/zendesk/maxwell/MaxwellParser.java

```diff
@@ -187,10 +187,17 @@ private MaxwellAbstractRowsEvent processRowsEvent(AbstractRowEvent e) throws Sch
 					break;
 				case MySQLConstants.QUERY_EVENT:
 					QueryEvent qe = (QueryEvent) v4Event;
-					// TODO: need to handle this case.
-					// The MySQL guys say some storage engines will output a "COMMIT" QUERY_EVENT at the
-					// end of the stream.
-					throw new RuntimeException("Unhandled QueryEvent: " + qe);
+
+					if ( qe.getSql().toString().equals("COMMIT") ) {
+						// some storage engines(?) will output a "COMMIT" QUERY_EVENT instead of a XID_EVENT.
+						// not sure exactly how to trigger this.
+						if ( !list.isEmpty() )
+							list.getLast().setTXCommit();
+
+						return list;
```

Comment:
A suggestion:

Instead of this xid stamping at the end of commit, can we
1. Add a maxwell_xid when we see transactions
2. At the end, associate mysql_xid with maxwell_xid when we see XID_EVENT. (So if someone wants mysql_xid, they can search for maxwell_xid which matches it)

This way, people still have ways of grouping up transactions, and also, we don't have to do this buffering thing.